app/src/main/java/com/example/login/presentation/LogInFragment.kt

```diff
@@ -0,0 +1,74 @@
+package com.example.login.presentation
+
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Toast
+import androidx.fragment.app.activityViewModels
+import androidx.navigation.Navigation
+import com.example.login.viewmodel.UserViewModel
+import com.example.login.R
+import com.example.login.databinding.FragmentLogInBinding
+import java.util.regex.Matcher
+import java.util.regex.Pattern
+
+class LogInFragment : Fragment() {
+    private lateinit var binding: FragmentLogInBinding
+    private val userViewModel: UserViewModel by activityViewModels()
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        binding = FragmentLogInBinding.inflate(inflater, container, false)
+
+        binding.btnLogIn.setOnClickListener {
+            val login = binding.edLogin.text.toString()
+            val password = binding.edPass.text.toString()
+            logInValidation(password = password, login = login)
+        }
+        return binding.root
+    }
+
+    private fun loginSuccessful(login: String) {
+        Toast.makeText(context, R.string.login_successful, Toast.LENGTH_SHORT).show()
+        view?.let { Navigation.findNavController(it).navigate(R.id.fromLogInToHello) }
+        userViewModel.userNameModel.value = login
+        binding.edPass.text.clear()
+    }
+
+    private fun message(content: Int) {
+        Toast.makeText(context, content, Toast.LENGTH_SHORT).show()
+    }
+
+    private fun logInValidation(password: String, login: String) {
```

Comment:
all validation should be in viewmodel - take a look in mvvm

Reply:
done